### PR TITLE
tf-idf, contractions, and unfrozen transformers

### PR DESCRIPTION
**Pull request by Jazhyc**
I added tf-idf weighting to the sentence embeddings and use the contractions from @nomomon's bachelor's project to improve pre-processing. Our validation accuracy with these changes is ~28% which seems to be lower compared to before.

Additionally, I added a notebook to test out fine-tuning of the BERT model which attains a 36% accuracy

Mainly, this pull request exists to see if a merge is possible

### CODE REVIEW

**Comment by nomomon** [comment]
this should not be merged at any cost
